src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java

```diff
@@ -462,11 +463,28 @@ public ItemEntity getOriginalEntity() {
         @Nonnull
         private final ItemStack crafting;
         private final IInventory craftMatrix;
+        @Nullable
+        private final IRecipe<?> recipe;
+
+        // TODO 1.17 REMOVE
         public ItemCraftedEvent(PlayerEntity player, @Nonnull ItemStack crafting, IInventory craftMatrix)
```

Comment:
Possibly mark this constructor as `@Deprecated`, to further discourage any current callers and alert them to change to the other constructor. Also, might be worth adding a small note that just remarks "see the other constructor".

Reply:
Done